File: reference_answer/avg_c/time.c
#include <sys/time.h>

long long currentTimeMillis() {
    long long tmp;
    struct timeval tv;
    gettimeofday(&tv, NULL);
    tmp = tv.tv_sec;
    tmp = tmp * 1000;
    tmp = tmp + (tv.tv_usec / 1000);
    return tmp;
}

File: reference_answer/avg_kn/src/nativeMain/kotlin/Time.kt
import kotlinx.cinterop.alloc
import kotlinx.cinterop.memScoped
import kotlinx.cinterop.ptr
import platform.posix.gettimeofday
import platform.posix.timeval

fun currentTimeMillis() = memScoped {
    val timeVal = alloc<timeval>()
    gettimeofday(timeVal.ptr, null)
    val sec = timeVal.tv_sec
    val usec = timeVal.tv_usec
    (sec * 1_000L) + (usec / 1_000L)
}
